### ISSUE TITLE
Admin Privileges Issue

### ISSUE THREAD

**Issue by Plinpod** [robb0wen/synthwave-vscode#146]
Hey! I'm on windows and Synthwave says "You must run VS code with admin priviliges in order to enable Neon Dreams." I am running VSCode with admin privileges but I still continue to get the error. Also using VSCode Remote SSH not sure if that has something to do with it.

Thanks.

**Comment by guoliang** [comment]
I got same issue, tried to open VScode with admin rights from shell, and by right click "Run as Administrator", and ran with the same issue that VScode needs to be run with admin privilegium

**Comment by the-unknown** [comment]
Same here.

**Comment by t0nylombardi331** [comment]
**tl;dr** 
I fixed the issue with 
`sudo chmod ugo+rw /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench`

I think the error messages could be a little clearer.  I did some digging and  I edited the extension with:  
```
  }
      } catch (e) {
        if (/ENOENT|EACCES|EPERM/.test(e.code)) {
          vscode.window.showInformationMessage(
            e +
              " You must run VS code with admin privileges in order to enable Neon Dreams."
          );
          return;
        }
```
`/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench` did not have read/write access. Therefore, it was not copying the theme_template.js to that directory nor writing it to workbench.html

**Comment by Plinpod** [comment]
Has anyone found a solution for this? Still running into this problem on Windows. Thanks!

**Comment by Plinpod** [comment]
I found out my issue here. I was connected to my Remote SSH inside VScode and it was install the theme on there. I had to disconnect to make sure the theme was installed locally and then it worked.

**Comment by Plinpod** [comment]
I noticed this is an issue when the you install the extension on a remote SSH through VScode.

**Comment by Golotvinov** [comment]
Same thing on Linux. Even if you give full admin rights to every single directory and folder that is somehow related to VS Code it will still tell you that you need Admin rights. Tried with chown, tried chmod 777 on everything VS Code related and nothing helped.

The following code works BUT it only launches a version of the application without user data. AKA you won't have your extensions / settings etc.
`sudo code --user-data-dir="~/.vscode-root"`

**Comment by Cyloo** [comment]
> I noticed this is an issue when the you install the extension on a remote SSH through VScode.

exact same problem here, did you find any solutions?

**Comment by jaitaiwan** [comment]
I just wanted to note that I'm also seeing this issue with VSCode remote over SSH

**Comment by eldritchgreene** [comment]
Same, would love to have a workaround for remote, maybe previous CSS method?

**Comment by mqt0029** [comment]
Here's how I do it. Tested on Ubuntu 20.04.1
1. Install VSCode from the .deb downloaded from [https://code.visualstudio.com/download](https://code.visualstudio.com/download)
2. Run VSCode as root with the following command
`sudo code --user-dir .config/Code`
3. Install theme, enable Neon Dream, reload VSCode to activate (VSCode will now throw the "Unsupported" error)
4. Install [Fix VSCode Checksum](https://marketplace.visualstudio.com/items?itemName=lehni.vscode-fix-checksums) and follow instruction.
5. Close VSCode and restart it with just `code` from terminal or from shortcut
6. Continue with extensions/settings sync

**Comment by xaviergmail** [comment]
I ran across this issue on Windows with VSCode running in WSL. I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.

**Comment by kevinmichaelchen** [comment]
@xaviergmail 
> I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.

I'm a Windows noob. How did you do this?

**Comment by Maycolx** [comment]
> @xaviergmail
> 
> > I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.
> 
> I'm a Windows noob. How did you do this?

only create new folder in disk c or d and create new archive.txt and open the archive on VSCODE and later procced configuration on enable the extension (ctrl + shift + p) "Dream" - Enable Neon Dream.

**Comment by Danble** [comment]
> > @xaviergmail
> > > I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.
> > 
> > 
> > I'm a Windows noob. How did you do this?
> 
> only create new folder in disk c or d and create new archive.txt and open the archive on VSCODE and later procced configuration on enable the extension (ctrl + shift + p) "Dream" - Enable Neon Dream.

Hehehe It works! Thanks, bro!

**Comment by MarcRodrigFelix** [comment]
> > > @xaviergmail
> > > > I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.
> > > 
> > > 
> > > I'm a Windows noob. How did you do this?
> > 
> > 
> > only create new folder in disk c or d and create new archive.txt and open the archive on VSCODE and later procced configuration on enable the extension (ctrl + shift + p) "Dream" - Enable Neon Dream.
> 
> Hehehe It works! Thanks, bro!

Wait! Help me out here please? I just got a pc so super new to the file tree and the environment. lol. How did you do this?

**Comment by jacadev** [comment]
> > @xaviergmail
> > > I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.
> > 
> > 
> > I'm a Windows noob. How did you do this?
> 
> only create new folder in disk c or d and create new archive.txt and open the archive on VSCODE and later procced configuration on enable the extension (ctrl + shift + p) "Dream" - Enable Neon Dream.

Thanks a lot dude!

**Comment by Maxtab** [comment]
> > @xaviergmail
> > > I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.
> > 
> > 
> > I'm a Windows noob. How did you do this?
> 
> only create new folder in disk c or d and create new archive.txt and open the archive on VSCODE and later procced configuration on enable the extension (ctrl + shift + p) "Dream" - Enable Neon Dream.

Another thanks message to say this still works on Dec. 2021

**Comment by guoliang** [comment]
doesn't work for me.
btw, follow this guide if you are getting notification from VScode saying that your installation has been corrupted
https://stackoverflow.com/questions/61028032/installation-appears-to-be-corrupt-unsupported-visual-studio-code

**Comment by giovanidevfullstack** [comment]
> > @xaviergmail
> > > I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.
> > 
> > 
> > I'm a Windows noob. How did you do this?
> 
> only create new folder in disk c or d and create new archive.txt and open the archive on VSCODE and later procced configuration on enable the extension (ctrl + shift + p) "Dream" - Enable Neon Dream.

Dude you're a genius!

**Comment by gnzlplcs** [comment]
> `sudo chmod ugo+rw /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench`

This solution works for me. Thanks so much!

**Comment by ibrahimxcool** [comment]
Try running vscode as an administrator

**Comment by JediRhymeTrix** [comment]
> **tl;dr** I fixed the issue with `sudo chmod ugo+rw /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench`
> 
> I think the error messages could be a little clearer. I did some digging and I edited the extension with:
> 
> ```
>   }
>       } catch (e) {
>         if (/ENOENT|EACCES|EPERM/.test(e.code)) {
>           vscode.window.showInformationMessage(
>             e +
>               " You must run VS code with admin privileges in order to enable Neon Dreams."
>           );
>           return;
>         }
> ```
> 
> `/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench` did not have read/write access. Therefore, it was not copying the theme_template.js to that directory nor writing it to workbench.html

I started getting this error today but I'm using the Insiders version of VSCode and I don't see workbench inside electron-browser. Any suggestions?

**Comment by ODYYYYYY** [comment]
> > @xaviergmail
> > > I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.
> > 
> > 
> > I'm a Windows noob. How did you do this?
> 
> only create new folder in disk c or d and create new archive.txt and open the archive on VSCODE and later procced configuration on enable the extension (ctrl + shift + p) "Dream" - Enable Neon Dream.

I have tried this method but it still didn't work ㄒoㄒ

**Comment by ODYYYYYY** [comment]
![image](https://user-images.githubusercontent.com/110674923/183081545-19a5df03-458e-46c1-93a4-57118a0b5c28.png)
My VS code is running in admin mode but I still can't enable the glow effect.

**Comment by pydjdev78** [comment]
first try to enable neon dreams in vscode on windows then switch to wsl, autometically will enable neon dreams. by editing extension.js file in your windows. follow this link, this worked for me. [https://stackoverflow.com/questions/73266964/why-cant-synthwave-84-enable-neon-dreams-on-windows](url)

**Comment by shelar1423** [comment]
still unable to get this done, someone help me please

**Comment by shelar1423** [comment]
> I ran across this issue on Windows with VSCode running in WSL. I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.

dint get u,im using windows it was working before but now after the update its getting the admi privilage issue

**Comment by Maxtab** [comment]
As referenced here, it is no longer working with the last VS Code update, but a fix should be released soon : https://github.com/robb0wen/synthwave-vscode/issues/263

**Comment by sangimed** [comment]
> I ran across this issue on Windows with VSCode running in WSL. I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.

Worked with Vscode+WSL under Windows 11 `22H2` 👍.

**Comment by ananthkamath** [comment]
> I ran across this issue on Windows with VSCode running in WSL. I managed to solve this by reopening VSCode in a local folder on Windows and enabling it, then switching back to WSL.

This works with VSCode and WSL for Windows 11, Thank you !

**Comment by bildungsroman** [comment]
Seeing this in Cursor now, every time a new version is installed. I'm on macOS

**Comment by nitroqwert** [comment]
> **tl;dr** I fixed the issue with `sudo chmod ugo+rw /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench`
> 
> I think the error messages could be a little clearer. I did some digging and I edited the extension with:
> 
> ```
>   }
>       } catch (e) {
>         if (/ENOENT|EACCES|EPERM/.test(e.code)) {
>           vscode.window.showInformationMessage(
>             e +
>               " You must run VS code with admin privileges in order to enable Neon Dreams."
>           );
>           return;
>         }
> ```
> 
> `/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench` did not have read/write access. Therefore, it was not copying the theme_template.js to that directory nor writing it to workbench.html


It isn't working for me and when I try to run the sudo command cmd just says "Command not found", so what could I do?
